lib/indexer/filterindexer.js

```diff
@@ -103,7 +123,20 @@ class FilterIndexer extends Indexer {
   async getFilterHeader(hash) {
     assert(hash);
 
-    return this.blocks.readFilterHeader(hash);
+    return this.blocks.readFilterHeader(hash, this.filterType);
+  }
+
+  /**
+   * Retrieve compact filter header by hash.
+   * @param {Hash} hash
+   * @returns {Promise} - Returns {@link Hash}.
+   */
+
+  async getFilterHash(hash) {
+    assert(hash);
+
+    const filterHash = await this.db.get(layout.f.encode(hash));
+    return filterHash;
```

Comment:
Can we do this here?
```suggestion
    return this.db.get(layout.f.encode(hash));
```